### ISSUE TITLE
Gonzales PE 4.2.3 Fails Parsing SCSS Custom CSS Variable With #{} Interpolation

### ISSUE THREAD

**Issue by dosentmatter** [tonyganch/gonzales-pe#290]
I am using [Sass Extract](https://github.com/jgranstrom/sass-extract) which uses [Gonzales PE ^4.2.2](https://github.com/jgranstrom/sass-extract/blob/master/package.json#L35). I am on Gonzales PE 4.2.3.

I'm trying to parse the [Bootstrap](https://github.com/twbs/bootstrap) scss to grab variables using Sass Extract. However it fails on one file, [`_root.scss`](https://github.com/twbs/bootstrap/blob/v4-dev/scss/_root.scss).

Here is the command I'm running in the Node repl

```
> sassExtract.renderSync({file: './node_modules/bootstrap/scss/bootstrap.scss'})
Thrown: Parsing error: Please check validity of the block starting from line #4

2 |   // Custom variable values only support SassScript inside `#{}`.
3 |   @each $color, $value in $colors {
4*|     --#{$color}: #{$value};
5 |   }

Syntax: scss
Gonzales PE version: 4.2.3
```

It fails on line 4:
```
    --#{$color}: #{$value};
```

`_root.scss` is the only file it fails on. If I [remove the import on this line](https://github.com/twbs/bootstrap/blob/v4-dev/scss/bootstrap.scss#L12), it works.

If I move the string inside the interpolation, it also works:
```
    #{--$color}: #{$value};
```

I know [custom css properties are supported](https://github.com/tonyganch/gonzales-pe/pull/276). But I don't know about custom css properties with interpolation. A workaround for now is to just skip this file since I'm only trying to grab the variables.

**Comment by rpearce** [comment]
Glad I found this issue, as I'm having the same problem with colors, as well:
```scss
@mixin iconColor($color, $colors...) {
  --icon-color: #{$color};
  @if length($colors) > 0 {
    @for $i from 1 through length($colors) {
      --icon-color-#{$i}: #{nth($colors, $i)};
    }
  }
}
```
It comes through [sass-lint](https://github.com/sasstools/sass-lint) as an error to like so:
```
error  Please check validity of the block starting from line #79  Fatal
```

**Comment by seventhqueen** [comment]
Hi, have you found any workaround for that?

**Comment by rpearce** [comment]
@seventhqueen Nope

**Comment by dosentmatter** [comment]
@seventhqueen, well if you are writing the Sass yourself, just move the hyphens inside the interpolation:
```
    --#{$color}: #{$value};
```
to
```
    #{--$color}: #{$value};
```

My problem is that I'm trying to run gonzales-pe on Sass out of my control.

**Comment by seventhqueen** [comment]
Thanks, will run some test. I have the freedom to change them

—
Gabriel
SeventhQueen Team
http://seventhqueen.com

Support site
http://seventhqueen.com/support


On 29 June 2018 at 07:59:56, Kevin Lau (notifications@github.com) wrote:

@seventhqueen <https://github.com/seventhqueen>, well if you are writing
the Sass yourself, just move the hyphens inside the interpolation:

    --#{$color}: #{$value};

to

    #{--$color}: #{$value};

My problem is that I'm trying to run gonzales-pe on Sass out of my control.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/tonyganch/gonzales-pe/issues/290#issuecomment-401247896>,
or mute the thread

**Comment by titos2k** [comment]
also it fails when variable has _$--name_  format.   For example Element vue UI library is using that format in their scss

**Comment by tonyganch** [comment]
Related issue: #197

**Comment by jdalton** [comment]
Good news! 

I've got this patched and working locally with an added unit test :tada:

I'd like to familiarize myself a bit more with the code base before pushing it forward.
Expect it to land within the week though.

**Update:**

I've opened a WIP PR at #295.

**Update:**

Closed by #295 :tada:

**Comment by jdalton** [comment]
**Update:**

I've opened a WIP PR #295.

**Comment by jdalton** [comment]
Closed by #295 :tada:

**Comment by webartov** [comment]
> also it fails when variable has _$--name_ format. For example Element vue UI library is using that format in their scss

The exact issue was not fixed.
Please see the comment with examples https://github.com/sasstools/sass-lint/issues/1248#issuecomment-475714471
